upstairs/src/downstairs.rs

```diff
@@ -649,6 +661,16 @@ impl Downstairs {
         client_id: ClientId,
         up_state: &UpstairsState,
     ) {
+        // Check whether we asked the IO task to stop ourselves
+        let stopped_due_to_fault = matches!(
+            self.clients[client_id].state(),
+            DsState::Stopping(ClientStopReason::Fault(..))
+        );
+
+        // Restart the IO task for that specific client, transitioning to a new
+        // state.
+        self.clients[client_id].reinitialize(up_state, self.can_replay);
```

Comment:
noticing that there's a slight behaviour change here: `reinitialize` now runs before `skip_all_jobs`